docs/concepts/architecture/cloud-controller.md

```diff
@@ -0,0 +1,250 @@
+
+ 1. Node Controller
+ 2. Volume Controller
+ 3. Route Controller
+ 4. Service Controller
+
+The CCM currently (v1.8) runs the following controllers from the above list
+
+1. Node Controller
+2. Route Controller 
+3. Service Controller
+
+Additionally, it runs another controller called the PersistentVolumeLabels controller. This controller is responsible for setting the zone and region labels on PersistentVolumes created in GCE and AWS clouds. 
+
+To summarize, the CCM runs the following 4 controllers:
+
+1. Node Controller
+2. Route Controller 
+3. Service Controller
+4. Persistent Volume Labels Controller
+
+**Note:** Volume Controller was deliberately chosen to *NOT* be a part of CCM. Due to the complexity involved and due to the existing efforts to abstract away vendor specific volume logic, it was decided that Volume Controller will not be moved to CCM.   
+
+The original plan to support volumes using CCM was to use Flex volumes to support pluggable volumes. However, a competing effort known as CSI is being planned to replace Flex. 
+
+Considering these dynamics, we decided to have an intermediate stop gap measure until CSI becomes ready.
+
+Work is in progress by the cloud provider working group (wg-cloud-provider) to enable PersistentVolume support using CCM - [kubernetes/kubernetes#52371](https://github.com/kubernetes/kubernetes/pull/52371).
+
+
+Functions of CCM
+------------------
+
+CCM inherits its functions from cloud provided dependent components of Kubernetes. This section is structured based on the components from which CCM inherits its functions.
+
+### 1. Kube Controller Manager
+
+The majority of CCM's functions are derived from KCM. As mentioned in the previous section, CCM runs the following control loops
+
+1. Node Controller
+2. Route Controller 
+3. Service Controller
+4. Persistent Volume Labels Controller
+
+#### Node Controller
+
+The Node Controller is responsible for initializing a node by obtaining information about the nodes running in the cluster from the cloud provider. The node controller performs the following functions
+
+1. Initialize a node with cloud specific zone/region labels
+2. Initialize a node with cloud specific Instance details (type, size etc.)
+3. Obtain the node's network addresses (IP addresses, hostname etc.)
+4. In case a node becomes unresponsive, then it checks the cloud to see if the node has been deleted from the cloud. It proceeds to delete the node object from Kubernetes if the node was indeed deleted from the cloud.  
+
+#### Route Controller
+
+The Route Controller is responsible for configuring routes in GCE appropriately so that containers on different nodes in the Kubernetes cluster can communicate with each other. The route controller is only applicable for GCE clusters.
+
```

Comment:
>configuring routes in GCE

should we change this to "configuring routes in the cloud" considering that this is about the generic architecture overview and this is *currently* only applicable to GCE clusters is mentioned in the same paragraph.